include/pf-applications/lac/solvers_linear.h

```diff
@@ -302,4 +303,62 @@ namespace LinearSolvers
     mutable MyTimerOutput timer;
   };
 
+  template <typename Operator>
+  class SolverDirectWrapper
+    : public LinearSolverBase<typename Operator::value_type>
+  {
+  public:
+    using VectorType      = typename Operator::vector_type;
+    using BlockVectorType = typename Operator::BlockVectorType;
+
+    SolverDirectWrapper(const Operator &op, SolverControl &solver_control)
+      : op(op)
+      , solver_control(solver_control)
+    {}
+
+    unsigned int
+    solve(VectorType &dst, const VectorType &src) override
+    {
+      return solve_internal(dst, src);
+    }
+
+    unsigned int
+    solve(BlockVectorType &dst, const BlockVectorType &src) override
+    {
+      VectorType src_, dst_;
+
+      const auto partitioner = op.get_system_partitioner();
+
+      src_.reinit(partitioner);
+      dst_.reinit(partitioner);
+
+      VectorTools::merge_components_fast(src, src_);
+      auto ret_val = solve_internal(dst_, src_);
+      VectorTools::split_up_components_fast(dst_, dst);
+
+      return ret_val;
+    }
+
+  private:
+    template <typename T>
+    unsigned int
+    solve_internal(T &dst, const T &src)
+    {
+      MyScope scope(timer, "direct::solve");
+
+      const auto &mtr = op.get_system_matrix();
+
+      TrilinosWrappers::SolverDirect solver(solver_control);
```

Comment:
Do we need to setup the solver during each solve or can it be reused?

Reply:
You are quite right, it can be reused. I made it a class member.